### PR TITLE
Docs/update c4 diagrams

### LINKED ISSUE #87: Re-do C4 diagrams

### PR DESCRIPTION
**Pull request by K-Beicher**
## Description

- This PR updates the two high-level C4 documents, moving away from docker containers and into four separate web applications.

## Related Issues

Closes #87 

## Testing

- [x] Yes
- [ ] No, not needed (give a reason below)
- [ ] No, I need help writing them

## Reviewer Focus

This PR requires an in-depth review.

## Checklist

For general documentation:

- [x] Spell-check
    - [ ] US
    - [ ] UK
- [ ] Did the page(s) preview correctly on your machine without breaking
- [ ] New category words (keywords) (if any) added to the code snippet file

### CODE REVIEW

File: images/context.puml [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 System(ui, "Data Resource structured by Seedcase")
 System_Ext(external_server, "External secure server")
 
-Rel(user_data_input, ui, "Uploads data")
-Rel(user_data_request, ui, "Find and request data for project")
-Rel(user_results, ui, "Explore findings of study")
+Rel(user_input, ui, "Uploads data")
+Rel(user_request, ui, "Find and request data for project")
 Rel(user_admin, ui, "Manages issues and requests for data projects")
+Rel(user_results, ui, "Explore findings of study")
```
Comment (signekb) [inline images/context.puml:23]:
```suggestion
Rel(user_results, ui, "Explore variables and projects")
```
Wanted to add that the shopping card will include that they can explore what variables are within the data resource - but not make it sound like everyone can explore the actual data. Not sure that I succeeded.

File: images/c4_container.puml
Comment (signekb) [inline images/c4_container.puml:1]: From the [C4 documentation](https://c4model.com/#ContainerDiagram) it seems that the users and external systems are still included in the container diagram, so maybe add that here as well? :)

File: images/c4_container.puml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Container(seed, "Main Seedcase", "web-interface")
+Container(sec, "User Authentication", "interface", "called when selecting any app which needs security")
+Container(sprout, "Sprout", "web-app", "uploads the research instance data")
+Container(flower, "Flower", "web-app", "makes metadata available to browse")
+Container(prop, "Propagate", "web-app", "allows others to request data")
+Container(garden, "Garden", "web-app", "handles workflow for data requests")
+Container_Boundary(backend, "data storage"){
+  ContainerDb(data, "Research Data", "db, csv, etc")
+  ContainerDb(user, "User Data", "db")
+}
```
Comment (signekb) [inline images/c4_container.puml:13]:
```suggestion
Container_Boundary(seed, "Seedcase Data Resource", "web-interface") {
	Container(sec, "User Authentication", "interface", "called when selecting any app which needs security")
	Container(sprout, "Sprout", "web-app", "uploads the research instance data")
	Container(flower, "Flower", "web-app", "makes metadata available to browse")
	Container(prop, "Propagate", "web-app", "allows others to request data")
	Container(garden, "Garden", "web-app", "handles workflow for data requests")
	Container_Boundary(backend, "data storage"){
  	ContainerDb(data, "Research Data", "db, csv, etc")
  	ContainerDb(user, "User Data", "db")
	}
}
```
Change "Main Seedcase" to a Container boundary and call it "Seedcase Data Resource" to fit with the Context diagram? 
And then change the relationships accordingly, so User authentication goes to Sprout, for instance.

**Review by signekb — DISMISSED** [review]
Very nice! I have some suggestions :)

File: images/c4-container.puml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
+
+Person(adm, "Administrator")
+Person(inp, "Data Contributor")
+Person(req, "Data Requester")
+
+Person_Ext(ext, "External User")
+
+Container_Boundary(seed, "Seedcase Data Resource", "web-interface") {
+	Container(land, "Main Seedcase page", "navigation page", "Main page showing installed apps")
```
Comment (signekb) [inline images/c4-container.puml:12]: @lwjohnst86 How do you imagine users navigating the different apps?

Comment (K-Beicher) [reply images/c4-container.puml:12]: As discussed we'll stick with the landing page for now.

**Comment by signekb** [comment]
They now look like this: 

## Context
image

## Container
image

**Comment by signekb** [comment]
@lwjohnst86 I couldn't find a way to remove the boundaries (a.k.a. the `[]`)  from the legend, so I kept them there for now.

File: software/quality-requirements.qmd [hunk 4]
```diff
@@ -1,87 +1,87 @@
...
+
+The system must have the functions that the users need. The ISO standard
+talks about completeness, correctness, and appropriateness which are all
+important aspects of functional suitability. We want all Seedcase
+products to have all the functions that are needed to store, organise,
+and retrieve research data. We want Seedcase to store data in a way that
+it stays correct, and is easy to maintain, and we want Seedcase to be
+have the appropriate features for the user.
+
+### Interaction Capability
```
Comment (lwjohnst86) [inline software/quality-requirements.qmd:50]:
```suggestion
### Interaction capability

```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**